src/db_to_vcf.py

```diff
@@ -0,0 +1,115 @@
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# pytest-workflow is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with gnomad-merge.  If not, see <https://www.gnu.org/licenses/
+"""
+db_to_vcf.py
+~~~~~~~~~~~~
+
+Export database created with `create_db.py` to VCF format.
+
+:copyright: (c) 2018 Sander Bollen
+:copyright: (c) 2018 Leiden University Medical Center
+
+:license: AGPL-3.0-or-later
+"""
+import argparse
+from pathlib import Path
+from typing import Tuple, Optional, Iterator
+import sqlite3
+
+import tqdm
+
+
+VCF_HEADER = ('##fileformat=VCFv4.2\n'
+              '##FILTER=<ID=PASS,Description="All filters passed">\n'
+              '##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">\n'
+              '##FORMAT=<ID=AD,Number=R,Type=Integer,Description="Allelic depths for the ref and alt alleles in the order listed">\n'  # noqa
+              '##FORMAT=<ID=DP,Number=1,Type=Integer,Description="Read Depth">\n'  # noqa
+              '##FORMAT=<ID=GQ,Number=1,Type=Integer,Description="Genotype Quality">\n'  # noqa
+              '##FORMAT=<ID=PL,Number=G,Type=Integer,Description="Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification">\n'  # noqa
+              '##FILTER=<ID=RF,Description="Failed random forests filters (SNV cutoff 0.1, indels cutoff 0.2)">\n'  # noqa
+              '##FILTER=<ID=AC0,Description="Allele Count is zero (i.e. no high-confidence genotype (GQ >= 20, DP >= 10, AB => 0.2 for het calls))">\n'  # noqa
+              '##FILTER=<ID=InbreedingCoeff,Description="InbreedingCoeff < -0.3">\n'  # noqa
+              '##FILTER=<ID=LCR,Description="In a low complexity region">\n'
+              '##FILTER=<ID=SEGDUP,Description="In a segmental duplication region">\n'  # noqa
+              '##INFO=<ID=AC,Number=A,Type=Integer,Description="Allele count in genotypes, for each ALT allele, in the same order as listed">\n'  # noqa
+              '##INFO=<ID=AF,Number=A,Type=Float,Description="Allele Frequency among genotypes, for each ALT allele, in the same order as listed">\n'  # noqa
+              '##INFO=<ID=AN,Number=1,Type=Integer,Description="Total number of alleles in called genotypes">\n'  # noqa
+              '#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO')
+
+
+def create_connection(db_path: Path) -> sqlite3.Connection:
+    """Create DB connection"""
+    return sqlite3.connect(str(db_path))
+
+
+def calc_af(ac: int, an: int) -> Optional[float]:
+    """Calculate af"""
+    if an > 0:
+        return ac / an
+
```

Comment:
What does it return when `an == 0` ?
Can this be made explicit?

Reply:
The function returns None in that case.